index.bs

```diff
@@ -1194,9 +1197,15 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
         Please note that this is a greater relaxation of "same-origin" restrictions than what
         [=document.domain=]'s setter provides.
 
-        These restrictions on origin values apply to [=WebAuthn Clients=]. If other specifications mimic the [=Web Authentication API=] to make WebAuthn [=credentials=] usable on other platforms (e.g. native mobile applications), they might have different rules for binding a caller to a [=Relying Party Identifier=]. Such rules are outside the scope of this specification.
+        These restrictions on origin values apply to [=WebAuthn Clients=].
     </div>
 
+    Other specifications mimicing the [=web authentication api|WebAuthn API=] to enable WebAuthn [=public key credentials=] on non-Web platforms (e.g. native mobile applications), MAY define different rules for binding a caller to a [=Relying Party Identifier=].
+
+    Such specifications MAY also define their own [=RP ID=] syntax, e.g., arbitrary URLs [[!URL]], or even arbitrary strings. [=Public key credentials=] created with such non-[=valid domain string=] [=RP ID=]s MUST be [=non-discoverable credentials=].
```

Comment:
I think we should remove the limitation to non-discoverable credentials because non-web cases will almost certainly want to use them (i.e. SSH). Conversely I think we should say that _only_ URLs are allowed. We can't technically enforce that but it avoids issues with them colliding with domain strings and nudging people in that direction might be useful.

Reply:
> it avoids issues with them colliding with domain strings

Is that necessarily a good thing, though? Under a restriction to URLs only, a website and a mobile app from the same developer, with the same backend server, would formally not be allowed to use the same RP ID. This seems to me like an unnecessary restriction, especially since it can't be enforced outside the web.

Reply:
How about we say something along the lines of:
```suggestion
    Such specifications MAY also define their own [=RP ID=] syntax, e.g., [=valid domain strings=] or arbitrary URIs [[!RFC3986]] [[!URL]]. 
```
ie, allows for URLs as well as URNs (which _are_ used in the FIM world)?

Reply:
actually, I take that back. ISTM that we oughta remove the limitation to non-discoverable credentials (to address @agl's first point), and regarding RP ID syntax, say that only valid domain strings or URIs are allowed (to address @emlun's point).